packages/components/divider/src/Divider.styles.ts

```diff
@@ -0,0 +1,97 @@
+import { tw } from '@spark-ui/internal-utils'
+import { cva, VariantProps } from 'class-variance-authority'
+
+import { intentVariants } from './divider/intents'
+
+export const dividerStyles = cva(['overflow-hidden'], {
+  variants: {
+    isEmpty: {
+      true: ['border-solid'],
+      false: ['inline-flex items-center', 'after:border-solid before:border-solid'],
+    },
+    orientation: {
+      vertical: ['w-fit inline-flex'],
+      horizontal: ['w-full'],
+    },
+    alignment: {
+      start: [],
+      end: [],
+      center: [],
+    },
+    intent: {
+      current: [],
+      outline: [],
+    },
+  },
+  defaultVariants: {
+    orientation: 'horizontal',
+    alignment: 'center',
+    intent: 'outline',
+  },
+  compoundVariants: [
+    {
+      isEmpty: true,
+      orientation: 'horizontal',
+      class: tw(['my-lg border-t-sm']),
+    },
+    {
+      isEmpty: true,
+      orientation: 'vertical',
+      class: tw(['mx-lg min-h-sz-24 border-l-sm']),
+    },
+    {
+      isEmpty: false,
+      orientation: 'horizontal',
+      class: tw([
+        'flex-row my-sm grow-0',
+        'before:border-t-sm',
+        'after:border-t-sm',
+        '[&>*]:px-lg',
```

Comment:
Why not having this style in `Divider.Content`?

Reply:
no, cuz in that case we need to pass the prop to the Content with a Provider and it increases the logic for that "simple" line many more. Thats the simplest way

Reply:
Yeap. Its 4 not having to drill down the orientation value from the divider to its content to indicate where the padding needs to be included. `asChild` + `children` combined make Provider the only way for sharing this setting between Content and its Parent Divider but it is too much complicated.